Added a section for the Prequery example. (#701)
mulesoft/docs-connectors: modules/ROOT/pages/salesforce/salesforce-composite-connector.adoc

Anypoint Connector for Salesforce Composite enables you to work with the Salesforce Batch and SObject Tree APIs. The connector exposes convenient methods for preparing subordinate requests, executing them in one batch, and parsing the results. Also it offers the posibility of creating SObject Trees through an operation separate from the batch ones, called createSobjectTree. This connector provides DataSense for preparing subordinate requests and parsing responses. DataSense is also provided for execution, and for creating SObject trees.

> Release Notes: xref:release-notes::connector/salesforce-composite-connector-release-notes-mule-4.adoc[Salesforce Composite Connector Release Notes] +

Reviewer: Should we update to full name, or is that done in the narrative directly following the title?

</dependency>
----

> Mule converts RELEASE to the latest version. To specify a version, view

Reviewer: ```suggestion
Mule converts `RELEASE` to the latest version. To specify a version, view
```
Is RELEASE syntax or is it a label or...? I'm just guessing here; maybe it does not require formatting? Where does it fit in the table of font choices in the style ref?

</mule>
----

> === HTTP Input For the Studio Example

Reviewer: ```suggestion
=== HTTP Input for the Studio Example
```

}
----

> == Pre-query Example

Reviewer: ```suggestion
== Pre-Query Example
```

> This example provides metadata for an object to be queried, generates two sub-requests that contain this metadata, and then executes the sub-requests. The following screenshots show the flow for the Pre-Query example:

Reviewer: GLOBALLY avoid using a hyphen after prefixes, per the style reference (see the exceptions):
```suggestion
This example provides metadata for an object to be queried, generates two subrequests that contain this metadata, and then executes the subrequests. The following screenshots show the flow for the Pre-Query example:
```

.The flow for this example includes an HTTP Listener component, Transform components, and multiple Salesfore Composite operations.
image::salesforce/salesforce-composite-prequery-flow.png[Pre-Query flow in Studio]

=== Start Creating the Flow by Adding HTTP Listener

`HTTP Listener` initiates the flow when it detects an event in the `executeFlow` path:

. Create a new Mule project in Studio.

> . In the Mule Palette view, search for *HTTP* and select the *Listener* operation.

Reviewer: GLOBAL: "Mule Palette" appears as a label in the Studion interface, if I remember correctly, so it should be formatted in bold:
```suggestion
. In the *Mule Palette* view, search for *HTTP* and select the *Listener* operation.
```

. Drag the *Listener* operation onto the canvas.

> . In the Listener properties window, set the *Path* field to `/executeFlow`.

Reviewer: If this window has a label, that label should be in bold; if this is just what the window IS, without a label, then STET.

=== Add the First Transform Message Component

This `Transform Message` component sets a name for the object to be created and passes the name to the `Pre create` operation:

> . In the Mule Palette view, search for *Transform Message*.

Reviewer: ditto *Mule Palette*

. Drag the *Transform Message* component onto the canvas, to the right of *Listener*.
. Enter the name `MyNewAccount`:
+
[source,dataweave,linenums]
----
%dw 2.0
%output application/json
---
{
   Name: "MyNewAccount"
}
----

=== Add the Pre Create Operation

> The `Pre create` operation provides metatdata for creating an object called `NewAccount`. It generates a sub-request for the `executeCompositeBatch` operation:

Reviewer: ditto subrequest

> . In the Mule Palette view, search for *Salesforce* and select the *Salesforce Composite Pre create* operation.

Reviewer: ditto

. Drag the *Pre create* operation to the right of *Transform Message*.

> . Click the green plus icon to the right of the *Connector configuration* field to access the Salesforce Composite global element configuration fields.

Reviewer: ```suggestion
. Click the green plus icon (*+*) to the right of the *Connector configuration* field to access the Salesforce Composite global element configuration fields.
```

. Complete the fields using variables, as follows:
+
.Complete fields in the General section to create a reusable global element for Salesforce Composite Connector.
image::salesforce/salesforce-composite-global-config.png[Salesforce Composite Connector global element configuration]
+

> . In the Pre create properties window, enter `Account` in the *Type* field.

Reviewer: ditto re. window label vs. function

> . In the YAML file `src/main/resources/properties/project_name_dev.yaml`, declare and enter values for the variables in the global element .

Reviewer: ```suggestion
. In the YAML file `src/main/resources/properties/project_name_dev.yaml`, declare and enter values for the variables in the global element.
```
Note use of lowercase "global element," because I think I saw "Global element" in a previous file? See Term Presentation list for guidance.

. Load the YAML file as a global element.

=== Add the Second Transform Message Component

This `Transform Message` component sets values for the `ID` and `Type` fields, and then passes these values to the `Pre query` operation:

. Drag a second *Transform Message* component to the right of *Pre create*.
. Enter the following name-value pairs:
+
[source,dataweave,linenums]
----
%dw 2.0
%output application/json
---
{
"Id": payload.Id,
"Fields" : ["Id", "Name"],
"Type": "Account"
}
----

=== Add the Pre Query Operation

> The `Pre query` operation provides metadata for an object to be queried. It generates a sub-request for the `Execute composite batch` operation:

Reviewer: ditto

. Drag a Salesforce Composite *Pre query* operation to the right of the second *Transform Message*.
. In the *Query* field, enter the following query:
+
`Select Name from Account WHERE Name LIKE '%:name %'`

=== Add the Third Transform Message Component

> This `Transform Message` operation passes the output of the `Pre query` operation to the `Execute composite batch` operation.

Reviewer: ```suggestion
This `Transform Message` operation passes the output of the `Pre query` operation to the `Execute composite batch` operation:
```

. Drag a third *Transform Message* component to the right of *Pre query*.
. Set the output to `payload`:
+
[source,dataweave,linenums]
----
%dw 2.0
%output application/java
---
[
payload
]
----

=== Add the Execute Composite Batch Operation

> The  `Execute composite batch` operation executes the sub-requests that the `Pre create` and `Pre query` operations created.

Reviewer: ditto

> Drag a Salesforce Composite *Execute composite batch* operation to the right of the third *Transform Message*.

Reviewer: "composite" intentionally init. capped?

> === Add the Fourth Transform Message Component

Reviewer: Were each of these Level-3 subtopics implied by the text in their container Level-2 topic?
I know they are not listed individually, which is not necessary, but I just want to verify with you that the summary of what the example shows is enough to prepare the audience for all these subtasks.

> This `Transform Message` operation converts the output to JSON format.

Reviewer: ```suggestion
This `Transform Message` operation converts the output to JSON format:
```

. Drag a fourth *Transform Message* component to the right of *Execute composite batch*.
. Set the output to `application/json`:
+
[source,dataweave,linenums]
----
%dw 2.0
%output application/json
---
payload
----

=== XML for the Pre-Query Example

> Paste this code into your XML editor to quickly load the flow for this example into your Mule app. If needed, change the values to reflect your environment.

Reviewer: ```suggestion
Paste this code into your XML editor to quickly load the flow for this example to your Mule app. If needed, change the values to reflect your environment.
```
